src/transforms/glossary.ts

```diff
@@ -27,11 +34,31 @@ export function addGlossaryDefinition(
 
         if (at == null) return
 
+        const fragment = Editor.fragment(editor, at)
+
+        const { selection } = editor
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let toInsert = fragment as any
+        if (selection) {
+            const path = Path.common(selection.anchor.path, selection.focus.path)
+            if (Path.equals(selection.anchor.path, selection.focus.path)) {
+                // eslint-disable-next-line no-empty-pattern
+                for (const {} of path.slice(0, path.length-1)) {
+                    toInsert = toInsert[0].children
+                }
+            } else {
+                // eslint-disable-next-line no-empty-pattern
+                for (const {} of path) {
+                    toInsert = toInsert[0].children
+                }
+            }
+        }
```

Comment:
This will discard any formatting which is around selection.

Instead do something like this

```ts
const fragment = unpackBlocks(editor, Editor.fragment(editor, at)

function unpackBlocks(editor: Editor, fragment: Descendant[]): Descendant[] {
    function* unpack(nodes: Descendant[]) {
        for (node in nodes) {
            if (Editor.isBlock(editor, node)) {
                yield* unpack(editor, node.children)
            } else {
                yield node
            }
        }
    }

    return Array.from(unpack(fragment))
}
```